File: src/test/java/sirius/db/text/PatternSplitProcessorTest.java
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text;

import org.junit.Test;

import java.util.regex.Pattern;

public class PatternSplitProcessorTest extends TokenProcessorTest {

    @Test
    public void testTokenizing() {
        assertExactTokenizing("a7b8$c",
                              new PatternSplitProcessor(Pattern.compile("[^a-z]"), false, true),
                              "a",
                              "b",
                              "c");
        assertExactTokenizing("  A  B-6:7C-",
                              new PatternSplitProcessor(Pattern.compile("[^a-zA-Z0-9]"), true, true),
                              "  A  B-6:7C-",
                              "A",
                              "B",
                              "6",
                              "7C");
    }
}

File: src/test/kotlin/sirius/db/text/PatternSplitProcessorTest.kt
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text

import org.junit.jupiter.api.Test
import java.util.regex.Pattern

/**
 * Tests the [PatternSplitProcessor].
 */
class PatternSplitProcessorTest : TokenProcessorTest() {

    @Test
    fun tokenizing() {
        assertExactTokenizing(
            "a7b8\$c",
            PatternSplitProcessor(Pattern.compile("[^a-z]"), false, true),
            "a",
            "b",
            "c"
        )
        assertExactTokenizing(
            "  A  B-6:7C-",
            PatternSplitProcessor(Pattern.compile("[^a-zA-Z0-9]"), true, true),
            "  A  B-6:7C-",
            "A",
            "B",
            "6",
            "7C"
        )
    }
}
